token/cli/src/main.rs

```diff
@@ -455,10 +455,17 @@ fn resolve_mint_info(
             .rpc_client
             .get_token_account(&token_account)?
             .ok_or_else(|| format!("Could not find token account {}", token_account))?;
-        Ok((
-            Pubkey::from_str(&source_account.mint)?,
-            source_account.token_amount.decimals,
-        ))
+        let source_mint = Pubkey::from_str(&source_account.mint)?;
+        if let Some(mint) = mint_address {
+            if source_mint != mint {
+                return Err(format!(
+                    "Source {:?} does not contain {:?} tokens",
+                    token_account, mint
+                )
+                .into());
+            }
+        }
```

Comment:
@t-nelson I put this in resolve_mint_info because I thought it might be useful for other methods, but let me know if you think I should move this back out into `command_transfer()` (re https://github.com/solana-labs/solana-program-library/issues/1479#issuecomment-804430134)